six/modules/c++/cphd/include/cphd/CPHDWriter.h

```diff
@@ -60,55 +63,104 @@ class CPHDWriter
+            size_t numThreads = 0,
+            size_t scratchSpaceSize = 4 * 1024 * 1024);
 
     /*
-     *  \func addImage
-     *  \brief Pushes a new image to the file for writing. This only works with
-     *         valid CPHDWriter data types:
+     *  \func write
+     *  \brief Writes the complete CPHD into the file.
+     *         This only works with valid CPHDWriter data types:
+     *              std:: ubyte* (for compressed data)
      *              std::complex<float>
      *              std::complex<sys::Int16_T>
      *              std::complex<sys::Int8_T>
      *
-     *  \param image The image to be added. This should be sized to match the
-     *         dims parameter.
-     *  \param dims The dimensions of the image.
-     *  \param vbmData The vector based metadata. This should have data
-     *         equal to: dims.row * metadata.data.getNumBytesVBM.
+     *  \param pathname The desired pathname of the file.
+     *  \param pvpBlock The vector based metadata to write.
+     *  \param widebandData .The wideband data to write to disk
+     *  \param pvpBlock The vector based metadata to write.
      */
-    template <typename T>
-    void addImage(const T* image,
-                  const types::RowCol<size_t>& dims,
-                  const sys::ubyte* vbmData);
+    template<typename T>
+    void write(
+            const std::string& pathname,
+            const PVPBlock& pvpBlock,
+            const T* widebandData,
+            const sys::ubyte* supportData = nullptr);
 
     /*
      *  \func writeMetadata
-     *  \brief Writes the header, metadata, and VBM into the file. This should
+     *  \brief Writes the header, and metadata into the file. This should
      *         be used in situations where you need to write out the CPHD
-     *         data in chunks. Otherwise you should use addImage and write.
-     *         This will internally set the number of bytes per VBP.
+     *         data in chunks.
      *
      *  \param pathname The desired pathname of the file.
-     *  \param vbm The vector based metadata to write.
-     *  \param classification The classification of the file. Optional
-     *         By default, CPHD will not be populated with this value.
-     *  \param releaseInfo The release information for the file. Optional
-     *         By default, CPHD will not be populated with this value.
+     *  \param pvpBlock The vector based metadata to write.
+     */
+    void writeMetadata(
+            const std::string& pathname,
+            const PVPBlock& pvpBlock);
+
+    /*
+     *  \func writeSupportData
+     *  \brief (Optional) Writes the specified support Array to the file
+     *         Does not include padding.
+     *
+     *  \param data A pointer to the start of the support array that
+     *        will be written to file
+     *  \param id The unique identifier of the support array
+     */
+    template <typename T>
+    void writeSupportData(const T* data,
```

Comment:
Debating whether to delete this or not.
Not sure if a valid use case would be writing a specific support file to disk (or to just write a block of data to disk instead as in writeSupportData(const T* data) line 132)
Also, as it is currently implemented, the user would also have to make sure to call this with support arrays in increasing order of offsets for this function (which I could add to the comments).

Reply:
I think if someone wants to look at a particular support array, they can just go through `SupportBlock::read()`, right? (Also, did you mean to have (Optional) in the \brief line here? I'm not sure what you mean there.)